core/network/corda-interop-app/interop-contracts/src/main/kotlin/com/weaver/corda/app/interop/contracts/AssetTransferContract.kt

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.weaver.corda.app.interop.contracts
+
+import com.weaver.corda.app.interop.states.AssetPledgeState
+import com.weaver.corda.app.interop.states.AssetClaimStatusState
+import com.weaver.corda.app.interop.states.NetworkIdState
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Contract
+import net.corda.core.contracts.requireSingleCommand
+import net.corda.core.contracts.requireThat
+import net.corda.core.contracts.StaticPointer
+import net.corda.core.transactions.LedgerTransaction
+import java.time.Instant
+import java.util.*
+
+/**
+ * AssetTransferContract defines the rules for managing a [AssetPledgeState].
+ *
+ */
+class AssetTransferContract : Contract {
+    companion object {
+        // Used to identify our contract when building a transaction.
+        const val ID = "com.weaver.corda.app.interop.contracts.AssetTransferContract"
+    }
+
+    /**
+     * A transaction is valid if the verify() function of the contract of all the transaction's
+     * input and output states does not throw an exception.
```

Comment:
A transaction is valid if an exception is not thrown by any of the `verify()` contract functions corresponding to the transactions' input and output states.

Reply:
Done.